apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/TimedBufferingConsumer.cs

```diff
@@ -24,8 +26,8 @@ namespace Google.Cloud.Diagnostics.Common
     /// </summary>
     internal class TimedBufferingConsumer<T> : IFlushableConsumer<T>
```

Comment:
Now I'm starting to see where some duplication could be removed. You could potentially have a FlushableConsumerBase that deals with the semaphore, and calls abstract methods, guaranteeing only to call them while holding the semaphore. What do you think?